hilbertcurve/hilbertcurve.py

```diff
@@ -112,17 +112,38 @@ def coordinates_from_distance(self, h: int) -> List[int]:
         # done
         return x
 
-    def distance_from_coordinates(self, x_in: List[int]) -> int:
+    def distance_from_coordinates(self, *cords: Union[Iterable[int], int]) -> int:
         """Return the hilbert distance for a given set of coordinates.
 
         Args:
-            x_in (list): transpose of h
+            *cords : transpose of h
                          (n components with values between 0 and 2**p-1)
+            A single list or tuple is accepted or a set of n individual coordinates, but not both:
+                distance_from_coordinates([x,y,z,...,etc])
+                distance_from_coordinates((x,y,z,...,etc))
+                distance_from_coordinates(x,y,z,...,etc)
```

Comment:
I'm not sure I like this much freedom in the input and I think `*` prepended arguments make things more confusing than they need to be. I think we should make the input an `Iterable[Iterable[int]]`

Reply:
That's a fair position to take, and the added complexity is regrettable.
I went with my current solution because all three felt like things that people might reasonably be expected to try if they didn't look at what argument the function accepts. Given the functions name (distance_from_coordinates not distance_from_point), I feel the intuitive thing to try, especially on 2 or 3-dimensional curves, is directly passing the arguments. That is certainly how I expected it to work and what I tried first. Tuples I felt should be supported because from a mathematical perspective parentheses are what are used to denote points in space, and it doesn't matter whether or not the input is mutable. And given that supporting all three adds very little overhead and doesn't sacrifice any functionality or previous formatting I didn't see a reason not too.

I assume the complexity you are talking about relates to the error checking and figuring out which type of argument has been passed. I don't think there is really a cleaner way to do it, there at least isn't one that is immediately obvious to me. Of course, what we really need to do it cleanly is overloading and the only python pseudo equivalent is the way I did it.

Depending on the level of concern towards backwards compatibility one solution that could solve both the complexity and the unintuitive parameter type would be to limit distance_from_coordinates to only individual coordinates but then create a class distance_from_point which accepts iterable objects, unpacks them, and then passes them to distance_from_coordinates. This would streamline the processing of * prepended arguments in distance_from_coordinates. Of course, this would require refactoring any code updating from an old version of this package.